### PR TITLE
fix javadoc & warning for platform refactoring

### PR DESCRIPTION
**Pull request by cdietrich**
fix javadoc & warning for platform refactoring
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/validation/AbstractValidatorConfigurationBlock.java [hunk 1]
```diff
@@ -99,7 +99,6 @@ private Composite createStyleTabContent(Composite folder) {
 
 	/**
 	 * This implementation returns Dialog settings which were bind in {@link org.eclipse.xtext.ui.DefaultUiModule}
-	 * normally {@link org.eclipse.ui.plugin.AbstractUIPlugin#getDialogSettings()}<br>
```
Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/validation/AbstractValidatorConfigurationBlock.java:102]: maybe we also can or should change severity to warning

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/validation/AbstractValidatorConfigurationBlock.java:102]: but i did not find an option

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/validation/AbstractValidatorConfigurationBlock.java:102]: just `org.eclipse.jdt.core.compiler.problem.invalidJavadocTagsDeprecatedRef=disabled`

Comment (kthoms) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/validation/AbstractValidatorConfigurationBlock.java:102]: I also do not see an option. I have raised feature request https://bugs.eclipse.org/bugs/show_bug.cgi?id=567968 for that.
For now, I would not remove the line, but maybe just change `@link` to `@code`.

Comment (kthoms) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/validation/AbstractValidatorConfigurationBlock.java:102]: I was too eager. The option `org.eclipse.jdt.core.compiler.problem.invalidJavadocTagsDeprecatedRef=disabled`does it.

(Uncheck option "Report deprecated references")

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/validation/AbstractValidatorConfigurationBlock.java:102]: with disabled i dont get a warning

Comment (kthoms) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/validation/AbstractValidatorConfigurationBlock.java:102]: Yes, that's the consequence. Maybe the options "Report non visible/deprecated references" should also get severity drop-downs?

Comment (kthoms) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/validation/AbstractValidatorConfigurationBlock.java:102]: Or we go with setting all to warning. We might oversee real errors, but ATM there are none and we just have to have an eye on new warnings. Removing the comment line is IMO no option.

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/validation/AbstractValidatorConfigurationBlock.java:102]: used code instead of link

**Review by kthoms — APPROVED** [review]

**Review by szarnekow — APPROVED** [review]

**Pull request merged**